File: RubyCivitas/lib/sorpresa_salircarcel.rb
module Civitas
  class Sorpresa_salircarcel < Sorpresa
    
    def initialize(mazo)
      
      @mazo = mazo
      
    end
    
    def salir_del_mazo

      @mazo.inhabilitar_carta_especial(self)
      
    end
    
    def usada
      
      @mazo.habilitar_carta_especial(self)
      
    end
    
    def aplicar_a_jugador(actual, todos)

      if jugador_correcto(actual, todos)

        informe(actual, todos)

        salvoconducto = false

        todos.each { |items|

          if items.tiene_salvoconducto

            salvoconducto = true

          end
        }

        unless salvoconducto

          todos[actual].obtener_salvoconducto(self)
          salir_del_mazo

        end
      end
    end
  
  end  
end

File: JavaCivitas/src/civitas/SorpresaSalirCarcel.java
package civitas;

import java.util.ArrayList;

public class SorpresaSalirCarcel extends Sorpresa{

    private MazoSorpresas mazo;

    SorpresaSalirCarcel(MazoSorpresas mazo_a, String texto){

        super(texto);
        mazo = mazo_a;

    }

    void salirDelMazo(){
        mazo.inhabilitarCartaEspecial(this);
    }

    void usada(){
        mazo.habilitarCartaEspecial(this);
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){

        if (jugadorCorrecto(actual, todos) && todos.get(actual) != null){

            super.informe(actual, todos);
            boolean salvoconducto = false;

            for(Jugador cada_j : todos){

                if(cada_j.tieneSalvoconducto()){

                    salvoconducto = true;

                }

            }
            if(!salvoconducto){

                todos.get(actual).obtenerSalvoconducto(this);
                salirDelMazo();

            }

        }

    }

}
